web-app-router/scenes/Portal/Teams/TeamId/Apps/AppId/SignInWithWorldId/page/ClientInformation/index.tsx

```diff
@@ -0,0 +1,222 @@
+"use client";
+import { Button } from "@/components/Button";
+import { DecoratedButton } from "@/components/DecoratedButton";
+import { Input } from "@/components/Input";
+import { TYPOGRAPHY, Typography } from "@/components/Typography";
+import { useResetClientSecretMutation } from "./graphql/client/reset-secret.generated";
+import { useCallback, useState } from "react";
+import { CopyIcon } from "@/components/Icons/CopyIcon";
+import { toast } from "react-toastify";
+import { LockIcon } from "@/components/Icons/LockIcon";
+import clsx from "clsx";
+import { Redirects } from "./Redirects";
+import { useFetchSignInActionQuery } from "./graphql/client/fetch-sign-in-action.generated";
+import * as yup from "yup";
+import { yupResolver } from "@hookform/resolvers/yup";
+import { useForm } from "react-hook-form";
+import { useUpdateSignInActionMutation } from "./graphql/client/update-sign-in-action.generated";
+
+const schema = yup.object({
+  privacy_policy_uri: yup.string().url("Must be a valid URL").optional(),
+  terms_uri: yup.string().url("Must be a valid URL").optional(),
+});
+
+type ClientInformation = yup.InferType<typeof schema>;
+
+export const ClientInformationPage = (props: {
+  appID: string;
+  teamID: string;
+  action: {
```

Comment:
```suggestion
  action: SignInActionQuery["action"][number]
```

Reply:
@andy-t-wang Use type from the generated file, this is one of the advantages of codegen for queries.

Reply:
@andy-t-wang Also let's think how we want to handle the case when there is no action. It's actually shouldn't be the case here, because sign in action should be created when you create an app. But I think we should handle this case from the UI point of view.
Two options there, as I can see: 
1. Mark action as optional here and adjust code right on this page to hide some elements to display page properly.
2. Simply catch an error it with error.tsx file. [Docs](https://nextjs.org/docs/app/building-your-application/routing/error-handling)

Reply:
Yeah I just pulled that server side action we don't use it anyways